@types/index.d.ts

```diff
@@ -0,0 +1,417 @@
+declare module "opentok-react-native" {
+  import React from "react";
+  import { ViewProps } from "react-native";
+
+  type Callback<T = void> = () => T;
+
+  type CallbackWithParam<P, T = void> = (param: P) => T;
+
+  type VideoSource = "screen" | "camera";
+
+  interface SessionConnectEvent {
+    sessionId: string;
+    connection: Connection;
+  }
+
+  interface SessionDisconnectEvent {
+    sessionId: string;
+  }
+
+  interface ConnectionCreatedEvent extends Connection {}
+
+  interface StreamCreatedEvent extends Stream {}
+
+  interface StreamDestroyedEvent extends Stream {}
+
+  interface SubscriberAudioLevelEvent {
+    audioLevel: number;
+    stream: Stream;
+  }
+
+  interface Stream {
+    name: string;
+    streamId: string;
+    hasAudio: boolean;
+    hasVideo: boolean;
+    sessionId: string;
+    connectionId: string;
+    width: number;
+    height: number;
+    videoType: VideoSource;
+    connection: Connection;
+    creationTime: string;
+  }
+
+  interface Connection {
+    creationTime: string;
+    data: string;
+    connectionId: string;
+    sessionId?: string;
+  }
+
+  interface StreamPropertyChangedEvent {
+    newValue: any;
+    oldValue: any;
+    changedProperty: "hasAudio" | "hasVideo" | "videoDimensions";
+    stream: Stream;
+  }
+
+  interface OTSessionViewProps extends ViewProps {
+    /**
+     * TokBox API Key
+     */
+    apiKey: string;
+
+    /**
+     * TokBox Session ID
+     */
+    sessionId: string;
+
+    /**
+     * TokBox token
+     */
+    token: string;
+
+    /**
+     * Used to define session options
+     */
+    options?: OTSessionSessionOptions;
+
+    /**
+     * Used to send a signal to the session
+     */
+    signal?: any;
+
+    /**
+     * Event handlers passed into the native session instance.
+     */
+    eventHandlers?: OTSessionEventHandlers;
+  }
+
+  interface OTSessionSessionOptions {
+    /**
+     * default is false
+     */
+    connectionEventsSuppressed?: boolean;
+
+    /**
+     * Android only - valid options are 'mediaOverlay' or 'onTop'
+     */
+    androidZOrder?: "mediaOverlay" | "onTop";
+
+    /**
+     * Android only - valid options are 'publisher' or 'subscriber'
+     */
+    androidOnTop?: "publisher" | "subscriber";
+
+    /**
+     * Android only - default is false
+     */
+    useTextureViews?: boolean;
+
+    /**
+     * Android only - default is false
+     */
+    isCamera2Capable?: boolean;
+
+    /**
+     * Android only - default is false
```

Comment:
Please replace it with:

https://tokbox.com/developer/sdks/js/reference/OT.html#initSession - ipWhitelist